pac4j-oidc/src/main/java/org/pac4j/oidc/profile/azuread/AzureAdProfile.java

```diff
@@ -46,4 +54,23 @@ public String getUsername() {
         return (String) getAttribute(AzureAdProfileDefinition.UPN);
     }
 
+    @Override
+    public boolean isExpired() {
+        try {
+            JWT jwt = this.getIdToken();
+            JWTClaimsSet claims = jwt.getJWTClaimsSet();
+            Date expiresOn = claims.getExpirationTime();
+
+            Calendar now = Calendar.getInstance();
+            now.add( Calendar.MINUTE, 5 );
```

Comment:
Why add 5 minutes for now?

Reply:
Use case when there is no buffer time.....
   id_token expires 10:00:00
current time is 9:59:59
   id_token is passed to the web app
   web app passes it_token to AzureAd 10:00:01 - sorry token has expired call fails.

Reply:
OK. I see, but isn't 5 minutes a bit too much? 10 seconds maybe?

Reply:
Ideally this value should be configured by the implementer.
I agree 5 minutes might be too long, I figured better safe than sorry. The end result is that my id_token gets refreshed every 55 minutes instead of every hour.
Depending on load/activity or other logic processes that may be written I feel 10 seconds may not be long enough for ALL cases.

Reply:
OK. I understand. As we have an `AzureAdProfileDefinition`, I think it should hold this delay and use it when building the `AzureAdProfile`...

Reply:
In a previous PR ( https://github.com/pac4j/pac4j/pull/1047 ) I was setting the config in the profile. It was not needed for that specific issue and was left out. Should that change be brought into this one, or is there a better way to bring in the delay configuration ?

Reply:
I don't think we should put the config in the profile. It should be the responsibility of the profile definition to know that and pass it when building the profile...